Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -153,7 +153,10 @@ public override void Close()
             {
                 var transactionRollbackStatus = SnowflakeDbConnectionPool.GetPooling() ? TerminateTransactionForDirtyConnectionReturningToPool() : TransactionRollbackStatus.Undefined;
                 
-                if (CanReuseSession(transactionRollbackStatus) && SnowflakeDbConnectionPool.AddSession(SfSession))
+                if ((CanReuseSession(transactionRollbackStatus) ||
+                    SfSession.StillRunningAsyncQueries()) &&

```

Comment:
why || ? shouldn't it be `CanReuseSession(transactionRollbackStatus) && !SfSession.StillRunningAsyncQueries() && SnowflakeDbConnectionPool.AddSession(SfSession)`?

Reply:
With `CanReuseSession(transactionRollbackStatus) && !SfSession.StillRunningAsyncQueries()`, is it possible that the session gets closed even if there are still async queries running? Like if the session can't be reused and there are still async queries being executed?

Reply:
Removed